### PR TITLE
markdown-it-py plugin anchors_ext

### PR DESCRIPTION
**Pull request by miteshashar**
Markdown plugin to modify the token stream produced by the mdit_py plugin heading_anchors to make the heading a permalink, instead of an appended permalink

### CODE REVIEW

File: funnel/utils/markdown/base.py [hunk 1]
```diff
@@ -280,6 +287,7 @@ def render(self, text: Optional[str]) -> Optional[Markup]:
         'footnote',
         'footnote_ext',  # Must be after 'footnote' to take effect
         'heading_anchors',
+        'anchors_ext',  # Must be after 'heading_anchors' to take effect
```
Comment (jace) [inline funnel/utils/markdown/base.py:290]: Should we call it `heading_anchors_fix`?

File: funnel/utils/markdown/mdit_plugins/anchors_ext.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def anchors_ext(state: StateInline) -> None:
+    prev_token = None
+
+    for token in state.tokens:
+        if prev_token is None:
+            prev_token = token
+            continue
+        if token.type == 'inline' and prev_token.type == 'heading_open':  # type: ignore
```
Comment (jace) [inline funnel/utils/markdown/mdit_plugins/anchors_ext.py:18]: `type: ignore[specific_type]` needed

File: funnel/utils/markdown/base.py (hunk 1 above)
Comment (miteshashar) [reply funnel/utils/markdown/base.py:290]: Yes, sure!
Was just following the same convention as `footnotes_ext`.

File: funnel/utils/markdown/mdit_plugins/anchors_ext.py (hunk 2 above)
Comment (miteshashar) [reply funnel/utils/markdown/mdit_plugins/anchors_ext.py:18]: Ok

Comment (miteshashar) [reply funnel/utils/markdown/mdit_plugins/anchors_ext.py:18]: Though, the failure was statement not reachable.

File: tests/unit/utils/markdown/data/headings.toml [hunk 3]
```diff
@@ -29,7 +29,12 @@ profiles = [ "basic", "document",]
 [config.custom_profiles.heading_anchors]
 preset = "default"
 options_update = {"breaks" = true}
-plugins = [ "heading_anchors",]
+plugins = [ "heading_anchors"]
```
Comment (jace) [inline tests/unit/utils/markdown/data/headings.toml:32]: Space mismatch on either side.

**Review by jace — APPROVED** [review]

**Pull request merged**